@xen-orchestra/lite/src/libs/xen-api.ts

```diff
@@ -283,4 +288,83 @@ export default class XenApi {
       poolRef,
     ]);
   }
+
+  get vm() {
+    type VmsRef =
+      | {
+          vmRef: XenApiVm["$ref"];
+          vmsRef?: undefined;
+        }
+      | {
+          vmRef?: undefined;
+          vmsRef: XenApiVm["$ref"][];
+        };
+
+    return {
+      start: ({ vmRef, vmsRef }: VmsRef) => {
+        const _vmsRef = vmsRef ?? [vmRef];
```

Comment:
You can use a default argument: `{ vmRef, vmsRef = [vmRef] }`.

Reply:
It cause typescript issue.
In this case it considers `vmsRef` as `(string | undefined)[]`